apps/studio/components/interfaces/SignIn/SignUpForm.tsx

```diff
@@ -1,36 +1,60 @@
-import HCaptcha from '@hcaptcha/react-hcaptcha'
-import { CheckCircle, Eye, EyeOff } from 'lucide-react'
-import { parseAsString, useQueryStates } from 'nuqs'
 import { useRef, useState } from 'react'
-import { toast } from 'sonner'
-import * as yup from 'yup'
 import { useRouter } from 'next/router'
+import { parseAsString, useQueryStates } from 'nuqs'
+import { CheckCircle, Eye, EyeOff } from 'lucide-react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import z from 'zod'
+import HCaptcha from '@hcaptcha/react-hcaptcha'
+import { toast } from 'sonner'
 
 import { useSignUpMutation } from 'data/misc/signup-mutation'
 import { BASE_PATH } from 'lib/constants'
-import { passwordSchema } from 'lib/schemas'
 import { buildPathWithParams } from 'lib/gotrue'
 import {
   AlertDescription_Shadcn_,
   AlertTitle_Shadcn_,
   Alert_Shadcn_,
   Button,
-  Form,
-  Input,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Form_Shadcn_,
+  Input_Shadcn_,
 } from 'ui'
 import PasswordConditionsHelper from './PasswordConditionsHelper'
-
-const signUpSchema = passwordSchema.shape({
-  email: yup.string().email().required().label('Email'),
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+
+const schema = z.object({
+  email: z.string({ required_error: 'Email is required' }).email('Must be a valid email'),
+  password: z
+    .string()
+    .min(1, 'Password is required')
+    .max(72, 'Password cannot exceed 72 characters')
+    .refine((password) => {
+      // Basic password validation - you can enhance this based on your requirements
+      const hasUppercase = /[A-Z]/.test(password)
+      const hasLowercase = /[a-z]/.test(password)
+      const hasNumber = /[0-9]/.test(password)
+      const hasSpecialChar = /[!@#$%^&*()_+\-=\[\]{};`':"\\|,.<>\/?]/.test(password)
+      const isLongEnough = password.length >= 8
+
+      return hasUppercase && hasLowercase && hasNumber && hasSpecialChar && isLongEnough
+    }, 'Password must contain at least 8 characters, including uppercase, lowercase, number, and special character'),
 })
```

Comment:
note that this was copied from the `ResetPasswordForm` which had previously migrated off of `yup-password`